lisp/ess-s-lang.el

```diff
@@ -741,22 +741,6 @@ underscore character."
 ;; -----  The user can always customize `ess-S-assign' ...
 (ess-toggle-S-assign 'force-to-S-assign)
 
```

Comment:
This shouldn't be deleted. I believe @mmaechler is using this for teaching.

Reply:
That's a weird function to have in the package. Should this be in `ess-site`? Looking into its contents, wasn't that function-outline thing moved to /old?

Reply:
IMO stuff like this should definitely not be in core files like ess-s-lang or ess-site. Perhaps we could rename ess-font-lock.el, which also contains personal settings, to something like ess-personal and put this function (and others like it) there? 

And yes, the function-outline thing no longer exists for most users.

Reply:
Let's wait for @mmaechler to tell us if he still needs that function. Moving it to ess-site is a reasonable compromise indeed. 

ess-font-lock is ancient. I would also favor moving it to `./old` but I am afraid plenty of people might still rely on those.

Reply:
Perhaps we should load `lisp/old/` files from ess-site, and as soon as someone complains something doesn't work we remove it for good? This way we can keep old stuff around but out of sight and we make no promise to keep them working.

Reply:
@lionel- One reason I'm in favor of removing stuff like this is because it's all just one more thing to check when we rename stuff or fix new byte compile warnings.

Can we just `make-obsolete` and then remove it in the subsequent release?

Reply:
I think we should remove all at once. Gradual removals are extremely unnoing because they will break people's setup every time they update ESS. We don't want that. So maybe indeed, old/unesed stuff should just go into `./lisp/old` and be loaded from there from ess-site. 

At some point we will make melpa work without ess-site, entirely based on autoloads.

Reply:
I'd pretty strongly vote for `make-obsolete`  -- fits what Emacs typically does and fits the quite consistent  deprecation -> defunctation  cycle (with sufficient waiting time in between) that we use in R.

Specifically as you asked:  I'm pretty sure I'm using  `ess-font-lock` quite extensively, am slightly less sure about the toggle-ess-assign / underscore issue.
I'm too busy these days, travelling all day Sunday finishing a talk to be held, and will not be able to look into this... and yes, I would be  really grateful if you don't remove these just now: I'm constantly working the github version of ESS and when it breaks during my preparation of talk, I'd be extremely unhappy... in particular as I know much less how to go back to a previous version than with svn where that was trivial for me.